### PR TITLE
Radio Sync Fix

### PR DESCRIPTION
**Pull request by Rxann**
I have tested, and with this change, the radios seem to sync up now.

### CODE REVIEW

File: src/hooks/protections/receive_net_message.cpp [hunk 1]
```diff
@@ -235,7 +236,7 @@ namespace big
 			switch (msgType)
 			{
 			case rage::eNetMessage::MsgScriptMigrateHost: return true;
-			case rage::eNetMessage::MsgRadioStationSyncRequest: return true;
+			case rage::eNetMessage::MsgRadioStationSyncRequest: return false;
```
Comment (gir489returns) [inline src/hooks/protections/receive_net_message.cpp:239]: Now we will be susceptable to OOM kicks. You have to address both scenarios.

**Review by gir489returns — CHANGES_REQUESTED** [review]

File: src/hooks/protections/receive_net_message.cpp (hunk 1 above)
Comment (Rxann) [reply src/hooks/protections/receive_net_message.cpp]: I will do this.

Comment (Yimura) [reply src/hooks/protections/receive_net_message.cpp]: To be fair, there was no comment at all for why that was implemented. There probably should've been a comment there explaining that it patches a certain crash/kick.

File: src/hooks/protections/receive_net_message.cpp [hunk 2]
```diff
@@ -85,6 +84,7 @@ namespace big
 
 		rage::eNetMessage msgType;
 		player_ptr player;
+		rate_limiter unk_player_radio_requests{5s, 2};
```
Comment (gir489returns) [inline src/hooks/protections/receive_net_message.cpp:87]: Why is this outside the scope of MsgRadioStationSyncRequest? Make it a static, too, otherwise the compiler will discard it.

File: src/hooks/protections/receive_net_message.cpp [hunk 3]
```diff
@@ -236,7 +236,19 @@ namespace big
 			switch (msgType)
 			{
 			case rage::eNetMessage::MsgScriptMigrateHost: return true;
-			case rage::eNetMessage::MsgRadioStationSyncRequest: return false;
+			case rage::eNetMessage::MsgRadioStationSyncRequest:
+			{
+				if (unk_player_radio_requests.process())
+				{
+					if (unk_player_radio_requests.exceeded_last_process())
```
Comment (gir489returns) [inline src/hooks/protections/receive_net_message.cpp:243]: Since this is for "all" people in the car, we should probably down the limit to 1-2 seconds. We need to test what happens if someone just presses next channel on the radio a ton of times.

**Review by gir489returns — CHANGES_REQUESTED** [review]

File: src/hooks/protections/receive_net_message.cpp (hunk 3 above)
Comment (Rxann) [reply src/hooks/protections/receive_net_message.cpp:244]: Ok, I have changed to 2s for now to be on the high side.

File: src/hooks/protections/receive_net_message.cpp [hunk 4]
```diff
@@ -2,16 +2,16 @@
...
 #include "gta_util.hpp"
 #include "hooking/hooking.hpp"
 #include "lua/lua_manager.hpp"
 #include "natives.hpp"
 #include "script/scriptIdBase.hpp"
 #include "services/players/player_service.hpp"
-#include "util/session.hpp"
 #include "util/chat.hpp"
-#include "gta/enums.hpp"
+#include "util/session.hpp"
```
Comment (gir489returns) [inline src/hooks/protections/receive_net_message.cpp:14]: Revert these unnecessary changes to the header.

File: src/hooks/protections/receive_net_message.cpp [hunk 5]
```diff
@@ -120,7 +120,7 @@ namespace big
 					if (g.session.log_chat_messages)
 						chat::log_chat(message, player, spam_reason, is_team);
 					g_notification_service.push("PROTECTIONS"_T.data(),
-                                      
+
```
Comment (gir489returns) [inline src/hooks/protections/receive_net_message.cpp:123]: Unnecessary whitespace changes.

File: src/hooks/protections/receive_net_message.cpp [hunk 6]
```diff
@@ -235,7 +235,21 @@ namespace big
...
+			case rage::eNetMessage::MsgRadioStationSyncRequest:
+			{
+				static rate_limiter unk_player_radio_requests{2s, 2};
+
+				if (unk_player_radio_requests.process())
+				{
+					if (unk_player_radio_requests.exceeded_last_process())
+					{
+						// Make a translation for this new OOM kick protection
+						g_notification_service.push_error("PROTECTIONS"_T.data(), "OOM_KICK"_T.data());
```
Comment (gir489returns) [inline src/hooks/protections/receive_net_message.cpp:247]: ...So are you gonna make a translation for this?

**Review by gir489returns — DISMISSED** [review]

File: src/hooks/protections/receive_net_message.cpp (hunk 6 above)
Comment (GreySideMM) [reply src/hooks/protections/receive_net_message.cpp:247]: Don't forget to call session::add_infraction(player, Infraction::TRIED_KICK_PLAYER).
Also: There might be no need for a different notification text, imo.

File: src/hooks/protections/receive_net_message.cpp (hunk 4 above)
Comment (GreySideMM) [reply src/hooks/protections/receive_net_message.cpp:14]: Serious question: What exactly is the problem with this and similar changes (i.e. whitespaces)? Merging problems / more potential work?

File: src/hooks/protections/receive_net_message.cpp (hunk 6 above)
Comment (Rxann) [reply src/hooks/protections/receive_net_message.cpp:247]: The current translation includes {} for a player name, and this pr handles the case where player is nullptr(not host) so we can't get the name of the player. We also can't call an infraction because we don't know the player.

File: src/hooks/protections/receive_net_message.cpp (hunk 4 above)
Comment (Rxann) [reply src/hooks/protections/receive_net_message.cpp:14]: These were made by the formatter, but i will revert

File: src/hooks/protections/receive_net_message.cpp (hunk 6 above)
Comment (gir489returns) [reply src/hooks/protections/receive_net_message.cpp:247]: > Don't forget to call session::add_infraction(player, Infraction::TRIED_KICK_PLAYER). Also: There might be no need for a different notification text, imo.

To who? We don't have a player pointer.

File: src/hooks/protections/receive_net_message.cpp (hunk 4 above)
Comment (gir489returns) [reply src/hooks/protections/receive_net_message.cpp:14]: > Serious question: What exactly is the problem with this and similar changes (i.e. whitespaces)? Merging problems / more potential work?

It's just unnecessary.

Comment (Yimura) [reply src/hooks/protections/receive_net_message.cpp:14]: The reordering of includes is fine as that's just a formatter thing to do.

File: src/hooks/protections/receive_net_message.cpp (hunk 5 above)
Comment (Yimura) [reply src/hooks/protections/receive_net_message.cpp:123]: Consequence of the formatter, those tabs shouldn't have been there to begin with.

**Review by Yimura — APPROVED** [review]

**Comment by Yimura** [comment]
I'm considering this finished, I'll wait a few hours before merging this in case it wasn't.

**Comment by Rxann** [comment]
> I'm considering this finished, I'll wait a few hours before merging this in case it wasn't.

Nope, if you think it's good it's good!

**Comment by gir489returns** [comment]
> I'm considering this finished, I'll wait a few hours before merging this in case it wasn't.

I was waiting on him to fix the whitespace changes, but if you want to merge it as is that's fine with me.

**Comment by Yimura** [comment]
> I was waiting on him to fix the whitespace changes, but if you want to merge it as is that's fine with me.

Those whitespace changes would've happened the next time someone touched the file with the formatter active (like me for example). It's important to differentiate between random newlines, line endings being changed and tabs being removed and includes being order alphabetically.
